File: leetcode/designHashMap.js
let myMap = []
let sz = 769
var MyHashMap = function() {
    for(let i=0; i<sz; i++) myMap[i] = []
};

/** 
 * @param {number} key 
 * @param {number} value
 * @return {void}
 */
MyHashMap.prototype.put = function(key, value) {
    let bucket = key % sz
    for(let i=0; i<myMap[bucket].length; i++) {
        if(myMap[bucket][i][0] === key){
            myMap[bucket][i][1] = value
            return
        }
    }
    
    myMap[bucket].push([key, value])
};

/** 
 * @param {number} key
 * @return {number}
 */
MyHashMap.prototype.get = function(key) {
    let bucket = key % sz
    for(let i=0; i<myMap[bucket].length; i++) {
        if(myMap[bucket][i][0] === key) return myMap[bucket][i][1]
    }
    return -1
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashMap.prototype.remove = function(key) {
    let bucket = key % sz
    for(let i=0; i<myMap[bucket].length; i++) {
        if(myMap[bucket][i][0] === key) {
            myMap[bucket][i] = myMap[bucket][myMap[bucket].length-1]
            myMap[bucket].pop()
        }
    }
};

/** 
 * Your MyHashMap object will be instantiated and called as such:
 * var obj = new MyHashMap()
 * obj.put(key,value)
 * var param_2 = obj.get(key)
 * obj.remove(key)
 */

File: leetcode/designHashMap.py
class MyHashMap:
    
    myMap = []
    sz = 769

    def __init__(self):
        self.myMap = self.sz*[None]

        
    def put(self, key: int, value: int) -> None:
        bucket = key % self.sz
        
        if(self.myMap[bucket] == None):
            self.myMap[bucket] = [[key, value]]
            return
        
        for i in range(0, len(self.myMap[bucket])):
            if(self.myMap[bucket][i][0] == key):
                self.myMap[bucket][i][1] = value
                return
        
        self.myMap[bucket].append([key, value])
        

    def get(self, key: int) -> int:
        bucket = key % self.sz
        
        if(self.myMap[bucket] == None): return -1
        
        for i in range(0, len(self.myMap[bucket])):
            if(self.myMap[bucket][i][0] == key): return self.myMap[bucket][i][1]
        
        return -1
    

    def remove(self, key: int) -> None:
        bucket = key % self.sz
        
        if(self.myMap[bucket] == None): return
        for i in range(0, len(self.myMap[bucket])):
            if(self.myMap[bucket][i][0] == key):
                self.myMap[bucket].pop(i)
                return


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)